src/wr_logic_teleop_arm/src/SolenoidController.cpp

```diff
@@ -0,0 +1,55 @@
+#include "SolenoidController.hpp"
+#include "ros/node_handle.h"
+#include "ros/subscriber.h"
+#include "std_msgs/Bool.h"
+
+#include <fcntl.h>
+#include <fstream>
+
+SolenoidController::SolenoidController(ros::NodeHandle& n) : 
+    extendYSub {n.subscribe("/hci/arm/gamepad/button/y", MESSAGE_QUEUE_LENGTH, 
+        &SolenoidController::extendSolenoid, this)}, 
+    yPressed {false}
+{
+    file.open("/sys/class/gpio/gpio6/value");
+    if (!file.is_open())
+    {
+        std::cout << "Unable to open /sys/class/gpio/gpio6/value";
+    }
+
+    fileExport.open("/sys/class/gpio/export");
+    if (!fileExport.is_open())
+    {
+        std::cout << "Unable to open /sys/class/gpio/export";
+    }
+    fileExport << pin;
```

Comment:
The first open will always fail, since you need to first export the pin for the `value` file to exist.  After you export, then you can open the `value` file.  It may also take some amount of time to wait for the file to export, so I recommend a structure like this:
```suggestion
    fileExport.open("/sys/class/gpio/export");
    if (!fileExport.is_open()) {
        std::cout << "Unable to open /sys/class/gpio/export";
    }
    fileExport << pin;

    while (access("/sys/class/gpio/gpio6/value", F_OK) != 0) {
        std::this_thread::sleep_for(std::chrono::milliseconds{100});
    }

    file.open("/sys/class/gpio/gpio6/value");
    if (!file.is_open()) {
        std::cout << "Unable to open /sys/class/gpio/gpio6/value";
    }
```
Make sure to include `<chrono>` and `<thread>`.  In the future, this should move to C++17 and use the standard `filesystem` library.

Reply:
Fixed in 6ef61e0.